nixos/modules/services/networking/rustical.nix

```
@@ -0,0 +1,170 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  inherit (lib)
+    getExe
+    maintainers
+    mkDefault
+    mkEnableOption
+    mkIf
+    mkOption
+    mkPackageOption
+    ;
+
+  inherit (lib.types)
+    attrsOf
+    bool
+    either
+    int
+    nullOr
+    path
+    port
+    str
+    submodule
+    ;
+
+  cfg = config.services.rustical;
+  settingsFormat = pkgs.formats.toml { };
+in
+{
+  options.services.rustical = {
+    enable = mkEnableOption "Rustical CalDAV and CardDAV server";
+    package = mkPackageOption pkgs "rustical" { };
+    settings = mkOption {
+      type = submodule {
+        freeformType = settingsFormat.type;
+        options.http.port = mkOption {
+          type = port;
+          default = 4000;
+          description = ''
+            Which port this service should listen on.
+          '';
+        };
+      };
+      default = { };
+      description = ''
+        Configuration for Rustical, written as TOML
+
+        Refer to the [documentation](https://lennart-k.github.io/rustical/installation/configuration) for options.
+      '';
+    };
+    environmentFile = mkOption {
+      type = nullOr path;
+      default = null;
+      example = "/run/secrets/rustical";
+      description = ''
+        An environment file as defined in {manpage}`systemd.exec(5)`.
+
+        Use this to inject secrets, e.g. database or auth credentials out of band.
+
+        Refer to the [documentation](https://lennart-k.github.io/rustical/installation/configuration/#environment-variables) for options.
+      '';
+    };
+    environment = mkOption {
+      type = attrsOf (either int str);
+      default = { };
+      description = ''
+        Environment variables passed to the service. Any config option name prefixed with RUSTICAL_ takes priority over the one in the configuration file.
+      '';
+      example = {
+        RUSTICAL_DATA_STORE__TOML__PATH = "asd";
```

```suggestion
        RUSTICAL_DATA_STORE__TOML__PATH = "asd";
```
Please use a realistic example